indico/modules/events/layout/controllers/menu.py

```diff
@@ -105,9 +109,30 @@ def _process(self):
                                     custom_title=self.entry.title is not None)
         form = form_cls(entry=self.entry, obj=defaults, event=self.event, **form_kwargs)
         if form.validate_on_submit():
+            old_title = self.entry.localized_title or self.entry.default_data.title
+            old_enable = self.entry.is_enabled
+            old_new_tab = self.entry.new_tab
+            old_link_url = self.entry.link_url
             form.populate_obj(self.entry, skip={'html', 'custom_title'})
+
+            log_fields = {'title': 'Title', 'is_enabled': 'Show', 'new_tab': 'New Tab'}
+            changes = {'title': (old_title, form.data['title']),
+                       'is_enabled': (old_enable, form.data['is_enabled']),
+                       'new_tab': (old_new_tab, form.data['new_tab'])}
```

Comment:
`changes = self.entry.populate_from_dict(...)` takes care of most of this.

PS: `form.data` creates a dict on every access, it's generally better to use `form.<fieldname>.data` to access data of fields when you don't actually need/want a dict with all the data

Reply:
Fixed, now it should be better.